### PR TITLE
Transformation function for /check_policies

### PR DESCRIPTION
**Pull request by KirtiGautam**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=6rMD0L9)

### CODE REVIEW

File: src/app/core/models/platform/platform-policy-expense.model.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+export interface PolicyExpense {
```
Comment (Dimple16) [inline src/app/core/models/platform/platform-policy-expense.model.ts:1]: `PlatformPolicyExpense`

File: src/app/core/models/platform/platform-policy-expense.model.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  spent_at: Date;
+  merchant?: string;
+  foreign_currency?: string;
```
Comment (Dimple16) [inline src/app/core/models/platform/platform-policy-expense.model.ts:4]: are all of the keys added as `?` optional?

File: src/app/core/services/policy.service.ts [hunk 3]
```diff
@@ -16,6 +17,54 @@ export class PolicyService {
     private approverPlatformApiService: ApproverPlatformApiService
   ) {}
 
+  transformTo(transaction): PolicyExpense {
```
Comment (Dimple16) [inline src/app/core/services/policy.service.ts:20]: add transaction type

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: src/app/core/models/platform/platform-policy-expense.model.ts (hunk 2 above)
Comment (KirtiGautam) [reply src/app/core/models/platform/platform-policy-expense.model.ts:4]: Partial yes, all except `claim_amount` and `spent_at` are optional. Documented here: [Link](https://docs.fylehq.com/docs/fyle-platform-docs/4551411b95bdc-check-expense-for-policy-violation)

**Pull request merged**